### DISCUSSION TITLE
The llm node is not fetching the output from python node .

### DISCUSSION THREAD

**Discussion by dhananjay-nagpal01** [microsoft/promptflow#2222] [Q&A] [answered]
give me insights about why my llm node is not fetching output from my python file. whenever I provide some pdf in input and write prompt in hello.jinja2 file , it doen't answer my query. here is my flow:
$schema: https://azuremlschemas.azureedge.net/promptflow/latest/Flow.schema.json
environment:
  python_requirements_txt: requirements.txt
inputs:
  text:
    type: string
    default: C:\Users\dhananjay.nagpaul\Downloads\legal_doc_1.pdf
outputs:
  output:
    type: string
    reference: ${llm.output}
nodes:
- name: python_read
  type: python
  source:
    type: code
    path: python_read.py
  inputs:
    file_path: ${inputs.text}
- name: llm
  type: llm
  source:
    type: code
    path: hello.jinja2
  inputs:
    deployment_name: gpt-35-turbo
    model: gpt-3.5-turbo
    max_tokens: 256
    text: ${python_read}
  connection: to_replace_with_connection_name
  api: chat
- name: hello_prompt
  type: prompt
  source:
    type: code
    path: hello.jinja2
  inputs: {}. 

Here is the content of my python node I have created to read the pdf.
from promptflow import tool
import PyPDF2

# The inputs section will change based on the arguments of the tool function, after you save the code
# Adding type to arguments and return value will help the system show the types properly
# Please update the function name/signature per need
@tool
def my_python_tool(pdf_file_path: str) -> str:
    pdf_string = ""
    
    # Open pdf file in read binary mode
    pdf_file = open(pdf_file_path, 'rb')

    # Create PDF reader object
    pdf_reader = PyPDF2.PdfReader(pdf_file)

    # Read page by page
    for page_num in range(len(pdf_reader.pages)):
        page = pdf_reader.pages[page_num]

        # Extract text from page and add to string
        pdf_string += page.extract_text()

    # Close the pdf file
    pdf_file.close()

    return pdf_string

The llm  always gives me useless answers that it needs the pdf or it is trying to retrieve the pdf etc.  Please help me with this

**Comment by 0mza987** [comment, marked as answer]
In your llm node, please modify `text: ${python_read}` to `text: ${python_read.output}`.
Can refer to our example flow dag yamls:
[promptflow/examples/flows/standard/web-classification/flow.dag.yaml at main · microsoft/promptflow](https://github.com/microsoft/promptflow/blob/main/examples/flows/standard/web-classification/flow.dag.yaml)

**Reply by dhananjay-nagpal01** [reply]
I have changed that. It is yet now working. LLM is not taking any input from the python node.

**Reply by 0mza987** [reply]
What's the error message?

**Reply by 0mza987** [reply]
Looks like your input `file_path` in your node `python_read` does not match the real input name `pdf_file_path` in your python code. Try to make them the same. And I believe the error message is something like this:

> pf.flow.test failed with UserErrorException: TypeError: Execution failure in 'python_read': (TypeError) my_python_tool() got an unexpected keyword argument 'file_path'
